### DISCUSSION TITLE
how to load checkpoints?

### DISCUSSION THREAD

**Discussion by lezhang7** [mlfoundations/open_clip#499] [Q&A] [answered]
I have trained the clip based on this codebase, and I want to load the trained checkpoint to the original clip model by 
`import CLIP.clip as clip
model, preprocess = clip.load("checkpoints/best_model.pt", device=device, jit=False)`
how can to do it?

**Comment by rwightman** [comment]
@Magiccircuit not sure what lib you're using but doesn' lool like this one, this is OpenCLIP

Basic usage as per README to load a pretrained model https://github.com/mlfoundations/open_clip#usage

To load a local checkpoint you can change the `pretrained=` arg to point to a checkpoint path on your local machine. You can also create the model like in the example without a pretrained arg and to a normal torch.load().

**Reply by lezhang7** [reply]
I mean if there is a script in which I can transform the checkpoints from openclip model to openai's clip model, the code for the two models are slightly different, so I can't directly load checkpoint trained from openclip to openai's model

**Reply by rwightman** [reply]
You might be able to do it for a few of the models, but some details like activations differ even when they line up. Not sure what the point is though, OpenCLIP is a superset of OpenAI architecture wise. It can support the OpenAI models exactly but has other features that can be enabled and is more training / fine-tune friendly for use with AMP, etc.

**Comment by hamedrezazadeh** [comment, marked as answer]
@Magiccircuit 
I wonder if youre looking for this script to convert the trained checkpoint to Transformers:
[https://gist.github.com/rwightman/c79fd0241ed3c860e898114931c07990](https://gist.github.com/rwightman/c79fd0241ed3c860e898114931c07990)
[18831](https://github.com/huggingface/transformers/issues/18831)

**Reply by lezhang7** [reply]
thanks, this is great! Do you know if there are other scrip to convert openclip to openai clip? cause many work is based on openai clip model, while most popular training code is from openclip, the model configuration is slightly different for these two clip resoursces

**Reply by hamedrezazadeh** [reply]
nope,
maybe @rwightman could answer your question.
